### ISSUE TITLE
Enum values typization is not working as expected

### ISSUE THREAD

**Issue by NikitaYVolkov** [bameyrick/enum-keys-values-entries#5]
#### Expected result
Taking the `TrafficLight` enum example from #3, implicitly the enum looks like this:
```
enum TrafficLight {
    Stop = 0,
    Caution = 1,
    Proceed = 2,
}
```
And the expected results of Enum functions are the following:
```
Enum.keys(TrafficLight) === ['Stop', 'Caution', 'Proceed'] // satisfies string[]
Enum.values(TrafficLight) === [0, 1, 2] // satisfies TrafficLight[]
Enum.entries(TrafficLight) === [['Stop', 0], ['Caution', 1], ['Proceed', 2]] // satisfies [string, TrafficLight][]
```
In other words, the code below should have no errors:
```
let firstEnumValue = Enum.values(TrafficLight)[0]
firstEnumValue === 0; // true, since TrafficLight.Stop === 0
```
#### Actual result
In fact the returned types are the following:
```
(property) keys: <typeof TrafficLight>(_enum: typeof TrafficLight) => ("Stop" | "Caution" | "Proceed")[]
(property) values: <typeof TrafficLight>(_enum: typeof TrafficLight) => (typeof TrafficLight)[]
(property) entries: <typeof TrafficLight>(_enum: typeof TrafficLight) => ["Stop" | "Caution" | "Proceed", typeof TrafficLight][]
```
Where `TrafficLight` in `typeof TrafficLight` is an object, and the code below has no type errors:
```
let firstEnumValue = Enum.values(TrafficLight)[0]
firstEnumValue = {
    Stop: 0,
    Caution: 1,
    Proceed: 2,
}
```
#### Suggestion
The values of `TrafficLight` enum should be of `TrafficLight` type instead of `typeof TrafficLight`:
```
const sampleEnumValue: TrafficLight = 3 // Error: "Type '3' is not assignable to type 'TrafficLight'."
const anotherEnumValue: TrafficLight = 2 // no error, since TrafficLight.Proceed === 2
```
Following [this StackOverflow answer](https://stackoverflow.com/a/69160875/23208521) I suggest to change Enum function types to the following:
```
declare function enumKeys<E, K extends string>(_enum: { [key in K]: E }): Array<K>;
declare function enumValues<E, K extends string>(_enum: { [key in K]: E }): Array<E>;
declare function enumEntries<E, K extends string>(_enum: { [key in K]: E }): Array<[K, E]>;
```
Separate generics for enum keys and values lead to the following returned types:
```
(property) keys: <TrafficLight, "Stop" | "Caution" | "Proceed">(_enum: {
    Stop: TrafficLight;
    Caution: TrafficLight;
    Proceed: TrafficLight;
}) => ("Stop" | ... 1 more ... | "Proceed")[]
(property) values: <TrafficLight, "Stop" | "Caution" | "Proceed">(_enum: {
    Stop: TrafficLight;
    Caution: TrafficLight;
    Proceed: TrafficLight;
}) => TrafficLight[]
(property) entries: <TrafficLight, "Stop" | "Caution" | "Proceed">(_enum: {
    Stop: TrafficLight;
    Caution: TrafficLight;
    Proceed: TrafficLight;
}) => [...][]
```
With a single element of each returned array having the following types:
```
let keys_element: "Stop" | "Caution" | "Proceed"
let values_element: TrafficLight
let entries_element: ["Stop" | "Caution" | "Proceed", TrafficLight]
```
Which is as I expect them to be.

**Comment by bameyrick** [comment]
Sorry for misunderstanding your original issue, and thank you for providing so much detail. I've implemented what you suggested from the StackOverflow answer. The change should be available in [1.0.6](https://github.com/bameyrick/enum-keys-values-entries/releases/tag/1.0.6) and should hopefully resolves any typisation issues you were getting.

**Comment by NikitaYVolkov** [comment]
The typization is working like a charm now in 1.0.6 version, thank you for the brief reaction!
